xtask/src/generate/instructions.rs

```diff
@@ -0,0 +1,216 @@
+use convert_case::{Case, Casing};
+use std::{collections::BTreeMap, path::PathBuf};
+
+use crate::{generate::utils, stationpedia};
+
+pub fn generate_instructions(
+    stationpedia: &stationpedia::Stationpedia,
+    workspace: &std::path::Path,
+) -> color_eyre::Result<Vec<PathBuf>> {
+    let instructions_path = workspace
+        .join("ic10emu")
+        .join("src")
+        .join("vm")
+        .join("instructions");
+    if !instructions_path.exists() {
+        std::fs::create_dir(&instructions_path)?;
+    }
+    let mut writer =
+        std::io::BufWriter::new(std::fs::File::create(instructions_path.join("enums.rs"))?);
+    write_instructions_enum(&mut writer, &stationpedia.script_commands)?;
+
+    let mut writer =
+        std::io::BufWriter::new(std::fs::File::create(instructions_path.join("traits.rs"))?);
+
+    write_instruction_interface_trait(&mut writer)?;
+    for (typ, info) in &stationpedia.script_commands {
+        write_instruction_trait(&mut writer, (typ, info))?;
+    }
+    write_instruction_super_trait(&mut writer, &stationpedia.script_commands)?;
+
+    Ok(vec![
+        instructions_path.join("enums.rs"),
+        instructions_path.join("traits.rs"),
+    ])
+}
+
+fn write_instructions_enum<T: std::io::Write>(
+    writer: &mut T,
+    instructions: &BTreeMap<String, stationpedia::Command>,
+) -> color_eyre::Result<()> {
+    println!("Writing instruction Listings ...");
+
+    let mut instructions = instructions.clone();
+    for (_, ref mut info) in instructions.iter_mut() {
+        info.example = utils::strip_color(&info.example);
+    }
+
+    write!(
+        writer,
+        "use serde::{{Deserialize, Serialize}};\n\
```

Comment:
Technically, this should be serde_derive, see https://www.reddit.com/r/rust/comments/1602eah/comment/jxkciap/?utm_source=reddit&utm_medium=web2x&context=3

Reply:
I'm not sure I follow the repo's read-me or the reddit post...

Reply:
it's just a best practice that is not followed in the ecosystem currently, the idea is that if you use `serde_derive` for the proc macros, crates only using serde without derive can run earlier since it doesn't need syn+quote